common/namespace/namespace.go

```diff
@@ -299,6 +301,13 @@ func (ns *Namespace) CustomSearchAttributesMapper() CustomSearchAttributesMapper
 	return ns.customSearchAttributesMapper
 }
 
+func (ns *Namespace) GetWorkflowRules() []*rulespb.WorkflowRule {
+	if ns.config.WorkflowRules == nil {
+		return []*rulespb.WorkflowRule{}
```

Comment:
Why not `nil`? This is extra allocation, and empty slices and nil slices should be treated in same way.